proto/poktroll/supplier/event.proto

```diff
@@ -6,23 +6,50 @@ option (gogoproto.stable_marshaler_all) = true;
 
 import "cosmos_proto/cosmos.proto";
 import "gogoproto/gogo.proto";
+import "cosmos/base/v1beta1/coin.proto";
 
 import "poktroll/shared/supplier.proto";
 
+enum SupplierUnbondingReason {
+  ELECTIVE = 0;
+  BELOW_MIN_STAKE = 1;
+}
+
 // EventSupplierStaked is emitted with the commitment of the supplier stake message.
 message EventSupplierStaked {
   poktroll.shared.Supplier supplier = 1 [(gogoproto.jsontag) = "supplier"];
+  // The session end height of the session in which the supplier was staked.
+  int64 session_end_height = 2 [(gogoproto.jsontag) = "session_end_height"];
 }
 
-// EventSupplierUnstaked is emitted with the commitment of the supplier unstake message.
+// EventSupplierUnbondingBegin is emitted when an application unstake message
+// is committed, indicating that an application has begun unbonding.
 message EventSupplierUnbondingBegin {
   poktroll.shared.Supplier supplier = 1 [(gogoproto.jsontag) = "supplier"];
-  int64 unbonding_height = 2 [(gogoproto.jsontag) = "unbonding_height"];
+  SupplierUnbondingReason reason = 2 [(gogoproto.jsontag) = "reason"];
+  // The session end height of the session in which the supplier unbonding began.
+  int64 session_end_height = 3 [(gogoproto.jsontag) = "session_end_height"];
+  // The height at which supplier unbonding will end.
+  int64 unbonding_height = 4 [(gogoproto.jsontag) = "unbonding_height"];
 }
```

Comment:
This is confusing.

"UnbondingBegingEvent" as the "height when unbonding will end".

1. Confirming that the comment is correct?
2. Regardless of whether (1) is correct or not, let's make the variable name itself more explicit so there's no ambiguity.

E.g. `unbonding_expected_completion_height`

Reply:
Thanks for calling this out. I'm going with `unbonding_end_height` for consistency with application unbonding events and "session_end_height" field(s).